sdks/python/apache_beam/tests/pipeline_verifiers.py

```diff
@@ -48,3 +55,63 @@ def describe_mismatch(self, pipeline_result, mismatch_description):
     mismatch_description \
       .append_text("Test pipeline job terminated in state: ") \
       .append_text(pipeline_result.current_state())
+
+
+def retry_on_fileio_error(exception):
+  """Filter allowing retries on file I/O errors."""
+  if isinstance(exception, RuntimeError) or \
+          isinstance(exception, IOError):
+    # GCS I/O raises RuntimeError and local filesystem I/O
+    # raises IOError when file reading is failed.
+    return True
+  else:
+    return False
+
+
+class FileChecksumMatcher(BaseMatcher):
+  """Matcher that verifies file(s) content by comparing file checksum.
+
+  Use fileio to fetch file(s) from given path. Currently, fileio supports
+  local filesystem and GCS.
+
+  File checksum is a SHA-1 hash computed from content of file(s).
+  """
+
+  def __init__(self, file_path, expected_checksum):
+    self.file_path = file_path
+    self.expected_checksum = expected_checksum
+
+  @retry.with_exponential_backoff(num_retries=MAX_RETRIES,
+                                  retry_filter=retry_on_fileio_error)
+  def _read_with_retry(self):
+    """Read path with retry if I/O failed"""
+    source = TextFileSource(self.file_path)
```

Comment:
`TextFileSource` is deprecated and will be removed soon. Use `fileio.ChannelFactory` instead.